### PR TITLE
Add td3 with per

### PR DESCRIPTION
**Pull request by MrSyee**

### CODE REVIEW

File: algorithms/per/td3_agent.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    "PER_BETA": 0.4,
+    "PER_EPS": 1e-6,
+}
+
+
+class Agent(AbstractAgent):
+    """ActorCritic interacting with environment.
+
+    Attributes:
+        memory (ReplayBuffer): replay memory
```
Comment (Curt-Park) [inline algorithms/per/td3_agent.py]: Type miss

File: algorithms/per/td3_agent.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        noise (OUNoise): random noise for exploration
```
Comment (Curt-Park) [inline algorithms/per/td3_agent.py]: Normal Dist.

File: algorithms/per/td3_agent.py (hunk 1 above)
Comment (MrSyee) [reply algorithms/per/td3_agent.py]: Modifications completed

File: algorithms/per/td3_agent.py (hunk 2 above)
Comment (MrSyee) [reply algorithms/per/td3_agent.py]: Modifications completed

File: algorithms/td3/agent.py [hunk 3]
```diff
@@ -178,6 +181,14 @@ def update_model(
         # for logging
         total_loss = critic_loss1 + critic_loss2
 
+        # update target networks
```
Comment (Curt-Park) [inline algorithms/td3/agent.py]: should be inside the policy update statement.

File: algorithms/td3/agent.py [hunk 4]
```diff
@@ -160,6 +161,8 @@ def update_model(
         curr_returns = rewards + hyper_params["GAMMA"] * next_values * masks
         curr_returns = curr_returns.to(device).detach()
 
+        # n-step return
```
Comment (Curt-Park) [inline algorithms/td3/agent.py]: should be removed

File: algorithms/replay_buffer.py [hunk 5]
```diff
@@ -36,7 +36,11 @@ def __init__(
             demo (deque) : demonstration deque
 
         """
-        self.memory = deque(maxlen=buffer_size) if not demo else demo
+        self.memory: deque = deque(maxlen=buffer_size)
+
+        if demo:
```
Comment (Curt-Park) [inline algorithms/replay_buffer.py]: should write any detail for the code which is not currently available.

**Pull request merged**